### PR TITLE
Adjust swagger URL path names

### PR DESCRIPTION
**Pull request by dchiquit**
The Swagger page automatically generates the names of the URL path
variables for a route. We use a Django Rest Framework extension to
easily generate our URL tree, which generates very verbose path variable
names by default. Those variable names are also used as queryset
filters, so we cannot simply change them in the router.

I couldn't find a way to overwrite the variable name in Swagger
configurations, so I just tried to mitigate some of the verbosity.
Previously all names had `"parent_lookup_"` prepended; this has been
removed.

There was some ambiguity with how dandiset identifiers were named;
depending on the level of the API, they were addressed as `identifier`,
`dandiset__pk`, and `version__dandiset__pk`. I changed the `identifier`
to instead be `dandiset__pk` so at least there's a little bit of
consistency accross routes.
Screenshot from 2020-07-15 15-59-59

I really don't like this solution, but I couldn't find anything better. There is something in https://drf-yasg.readthedocs.io/en/stable/custom_spec.html that allows swagger customization, but it doesn't allow us to rename variables. The only thing that might work is to upstream changes to that library, but that might not even work considering we have automatically generated getters that we wouldn't be able to annotate.

### CODE REVIEW

File: publish/views/dandiset.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ class DandisetViewSet(ReadOnlyModelViewSet):
     # This is not strictly necessary (it defaults to 'pk'), but it clarifies
     # that the 'identifier' property should be used when forming URLs
```
Comment (brianhelba) [inline publish/views/dandiset.py:36]: Could you update this comment to explain the rationale for the naming?

**Comment by brianhelba** [comment]
Will things like https://github.com/dandi/dandi-publish/blob/612e7eb34fee15d7371d0cacfc3a8113fb943bf2/publish/views/version.py#L49 break?

**Comment by mgrauer** [comment]
Is it possible to do `dandiset_identifier` instead of `dandiset_pk`?

**Comment by dchiquit** [comment]
@mgrauer I can change that one variable to `dandiset_identifier`, but sadly it would only be reflected in the `GET /dandisets/{dandiset__identifier}/` endpoint; everything else still uses `dandiset__pk`. I opted for `dandiset__pk` everywhere so there's some consistency.
The problem is that we store the identifier as an int in the `id` field, then provide a calculated property `identifier` that does string parsing to create the 6 character `000001` style identifier. You can't query on properties though, so we can't use `dandiset__identifier` because it doesn't work when feed into a queryset filter.

**Comment by mgrauer** [comment]
Thanks for looking into it, let's leave it as dandiset_pk to be consistent. We can probably find somewhere to document this as part of the swagger or perhaps elsewhere to document the API usage.

**Comment by mgrauer** [comment]
Did you investigate @brianhelba 's question? I want to make sure you saw it.

**Comment by brianhelba** [comment]
My question was resolved.

**Comment by brianhelba** [comment]
@mgrauer This can and should be fixed more comprehensively, but it will involve subclassing some of `rest_framework_extensions`. For now, it only effects the display of names in swagger, but an actual API consumer (i.e. an HTTP client author) doesn't need to care about any of these names, since they're all path segments.

**Comment by brianhelba** [comment]
I just merged the latest `master` into this, to fix conflicts.

**Pull request merged**